packages/ceramic-common/src/utils/doctype-utils.ts

```diff
@@ -252,10 +252,14 @@ export class DoctypeUtils {
      */
     static async convertRecordToDTO(record: any, ipfs: Ipfs): Promise<any> {
         if (DoctypeUtils.isSignedRecord(record)) {
-            const linkedBlock = await ipfs.block.get(record.link)
+            let linkedBlock = await ipfs.block.get(record.link)
+            if (linkedBlock.data instanceof Buffer) {
+                // transform Buffer into Uint8Array
+                linkedBlock = new Uint8Array(linkedBlock.data.buffer)
+            }
             return {
                 jws: record,
-                linkedBlock: new Uint8Array(linkedBlock.data.buffer),
+                linkedBlock,
```

Comment:
This logic is incorrect.

Reply:
I think.

Reply:
oh, outdated.

Reply:
It should be fine for now, tests are passing. We'll see when we need to upgrade IPFS and discard the forked one.

Reply:
sorry, why do you think it's incorrect?

Reply:
I'll create another PR since this one is merged.

Reply:
the logic will definitely fail when upgraded. Should be something like: 
```js
const block = await ipfs.block.get(record.link)
let linkedBlock = block.data instanceof Uint8Array ? block.data : new Uin8Array(block.data.buffer)
```